dataflow/src/main/java/org/checkerframework/dataflow/analysis/AnalysisResult.java

```diff
@@ -443,4 +445,124 @@ public S getStoreAfter(Block block) {
+    }
+
+    /**
+     * Return a printed representation of a map with the same type as the {@code nodeValues} field.
+     *
+     * @param <V> the type of values in the map
+     * @param nodeValues a map to format
+     * @return a printed representation of the given map
+     */
+    public static <V> String nodeValuesRepr(Map<Node, V> nodeValues) {
+        if (nodeValues.isEmpty()) {
+            return "{}";
+        }
+        StringJoiner result = new StringJoiner(String.format("%n    "));
+        result.add("{");
+        for (Map.Entry<Node, V> entry : nodeValues.entrySet()) {
+            Node key = entry.getKey();
+            result.add(String.format("%s => %s", nodeRepr(key), entry.getValue()));
+        }
+        result.add("}");
+        return result.toString();
+    }
+
+    /**
+     * Return a printed representation of a node.
+     *
+     * @param n a node to format
+     * @return a printed representation of the given node
+     */
+    public static String nodeRepr(Node n) {
+        return String.format(
+                "%s [%s %s %s]",
+                n, n.getClass().getSimpleName(), n.hashCode(), System.identityHashCode(n));
+    }
+
+    /**
+     * Return a printed representation of a collection of nodes.
+     *
+     * @param nodes a collection of nodes to format
+     * @return a printed representation of the given collection
+     */
+    public static String nodeCollectionRepr(Collection<? extends Node> nodes) {
+        StringJoiner result = new StringJoiner(", ", "[", "]");
+        for (Node n : nodes) {
+            result.add(nodeRepr(n));
+        }
+        return result.toString();
+    }
+
+    /**
+     * Return a printed representation of a map with the same type as the {@code treeLookup} field.
+     *
+     * @param treeLookup a map to format
+     * @return a printed representation of the given map
+     */
+    public static String treeLookupRepr(Map<Tree, Set<Node>> treeLookup) {
+        if (treeLookup.isEmpty()) {
+            return "{}";
+        }
+        StringJoiner result = new StringJoiner(String.format("%n    "));
+        result.add("{");
+        for (Map.Entry<Tree, Set<Node>> entry : treeLookup.entrySet()) {
+            Tree key = entry.getKey();
+            String treeString = key.toString().replaceAll("[ \n\t]+", " ");
+            if (treeString.length() > 65) {
+                treeString = "\"" + treeString.substring(0, 60) + "...\"";
+            }
+            result.add(treeString + " => " + nodeCollectionRepr(entry.getValue()));
+        }
+        result.add("}");
+        return result.toString();
+    }
+
+    /** Checks representation invariants on this. */
+    public void checkRep() {
+        // Require that each node in treeLookup exists in nodeValues.
+        for (Map.Entry<Tree, Set<Node>> entry : treeLookup.entrySet()) {
+            for (Node n : entry.getValue()) {
+                if (!nodeValues.containsKey(n)) {
+                    sleep(100);
```

Comment:
I'm a bit confused why you would need that here. We throw `BugInCF` in many places, so if this is a general problem, should that be solved where `BugInCF` is output?

Reply:
That doesn't work.  I'm not sure why.  Without this, printf output is interspersed.  Understanding why would be useful, but is outside the scope of this pull request.

Reply:
Can you at least add a TODO comment here to remark why it is here? This is very surprising and should be highlighted and fixed.